submodules/k8s/variables.tf

```diff
@@ -9,12 +9,10 @@ variable "ssh_pvt_key_path" {
   description = "SSH private key filepath."
 }
 
-
 variable "bastion_user" {
   type        = string
   description = "ec2 instance user."
   default     = "ec2-user"
```

Comment:
nit: it just seems weird that we attempt to set consistent defaults N levels deep rather than make certain things just required.

this bastion username highlights said issue. if we override the username at the top but fail to pass it around, we end up with a broken provision. We might want to think about where defaults truly live.

Reply:
all other usage should have just been optional(string) except here.

Reply:
The problem is when the top-level is optional this still passes through `null` and the default on the submodule doesn't kick in.

```
+  bastion_user                 = var.bastion.username
```

Reply:
ah, the difference between null and not present. so the default here is meaningless.

Reply:
it just strikes me as odd that we say `string` not optional(string), so a null should either fail or pick the default.  Might be a bug.

Reply:
TIL about `nullable` - pushed 9819132673ad83dc622b87f93134505e02dad62d